kornia/augmentation/pc/base.py

```diff
@@ -0,0 +1,152 @@
+from typing import Any, Dict, Optional, Tuple
+
+from torch import float16, float32, float64
+
+from kornia.augmentation.base import _AugmentationBase
+from kornia.augmentation.utils import _validate_input_dtype
+from kornia.core import Tensor
+from kornia.geometry.boxes import Boxes
+from kornia.geometry.keypoints import Keypoints
+from kornia.utils import eye_like, is_autocast_enabled
+
+
+class AugmentationBasePC(_AugmentationBase):
+    r"""AugmentationBasePC base class for customized point cloud augmentation implementations.
+
+    AugmentationBasePC aims at offering a generic base class for a greater level of customization.
+    If the subclass contains routined matrix-based transformations, `RigidAffineAugmentationBasePC`
+    might be a better fit.
+
+    Args:
+        p: probability for applying an augmentation. This param controls the augmentation probabilities
+          element-wise for a batch.
+        p_batch: probability for applying an augmentation to a batch. This param controls the augmentation
+          probabilities batch-wise.
+        same_on_batch: apply the same transformation across the batch.
+
+    Note:
+        By default, :math:`(B, N, 3)` represent xyz only, :math:`(B, N, 6)` represents xyz and normal,
+        :math:`(B, N, 9)` represents xyz, normal, and rgb. Complex point cloud data may use our point cloud
+        data type.
+    """
+
+    def __init__(self, p: float = 0.5, p_batch: float = 1.0, same_on_batch: bool = False) -> None:
+        super().__init__(p=p, p_batch=p_batch, same_on_batch=same_on_batch, keepdim=True)
+
+    def validate_tensor(self, input: Tensor) -> None:
+        """Check if the input tensor is formatted as expected."""
+        _validate_input_dtype(input, accepted_dtypes=[float16, float32, float64])
+        if len(input.shape) != 3 or input.size(-1) < 3:
+            raise RuntimeError(f"Expect (B, N, C) and C >= 3. Got {input.shape}.")
+
+    def transform_tensor(self, input: Tensor) -> Tuple[Tensor, Optional[Tensor], Optional[Tensor], Optional[Tensor]]:
+        """Split a given tensor to [xyz, normal, rgb, others].
+
+        Of which xyz is required, while the rest are optional.
+        """
+        C = input.size(-1)
+        if C < 3:
+            raise RuntimeError(
+                f"Insufficient input channels. We expect at least (B, N, C) and C >= 3. Got {input.shape}."
+            )
+        if C == 3:
+            return input, None, None, None
+        if C < 6:
+            return input[..., :3], None, None, input[..., 3:]
+        if C == 6:
+            return input[..., :3], input[..., 3:], None, None
+        if C < 9:
+            return input[..., :3], input[..., 3:6], None, input[..., 6:]
+        if C == 9:
+            return input[..., :3], input[..., 3:6], input[..., 6:], None
+        return input[..., :3], input[..., 3:6], input[..., 6:9], input[9:]
```

Comment:
@edgarriba The fields for a point cloud object can be represented like above.

Also, I think we should support `.from_open3d` and `.to_opend3d`.
http://www.open3d.org/docs/release/python_api/open3d.geometry.PointCloud.html

Reply:
100% agree